AppWidget/glance-widget/src/main/java/com/example/android/glancewidget/TodoListGlanceWidget.kt

```diff
@@ -0,0 +1,157 @@
+        Column(
+            modifier = GlanceModifier
+                .fillMaxSize()
+                .background(color = Color.White)
+                .padding(8.dp)
+        ) {
+            val context = LocalContext.current
+            val prefs = currentState<Preferences>()
+            Text(
+                text = context.getString(R.string.glance_todo_list),
+                modifier = GlanceModifier
+                    .fillMaxWidth()
+                    .padding(8.dp),
+                style = TextStyle(fontWeight = FontWeight.Bold, fontSize = 18.sp),
+            )
+            CountToggled()
+            LazyColumn {
+                items(groceryStringIds) {
+                    val toggledString = context.getString(it)
+                    val checked = prefs[booleanPreferencesKey(toggledString)] ?: false
+                    CheckBox(
+                        text = toggledString,
+                        checked = prefs[booleanPreferencesKey(toggledString)] ?: false,
+                        onCheckedChange = actionRunCallback<CheckboxClickAction>(
+                            actionParametersOf(
+                                toggledKey to toggledString,
+                                toggledValue to !checked
+                            )
+                        ),
+                        modifier = GlanceModifier.padding(12.dp)
+                    )
+                }
+            }
+        }
+    }
+}
+
+private val toggledKey = ActionParameters.Key<String>("ToggledKey")
+private val toggledValue = ActionParameters.Key<Boolean>("ToggledValue")
+
+class CheckboxClickAction : ActionCallback {
+    override suspend fun onRun(context: Context, glanceId: GlanceId, parameters: ActionParameters) {
+        val toggledString = requireNotNull(parameters[toggledKey]) {
+            "Add $toggledKey parameter in the ActionParameters."
+        }
+        // TODO: Remove this parameter once the checked state can be retrieved through
+        // ToggleableStateKey.
+        val checked = requireNotNull(parameters[toggledValue]) {
+            "Add $toggledValue parameter in the ActionParameters."
+        }
+
+        // TODO: Uncomment this once the checked state is retrieved from the action parameters
+        // in a lazy list
+//        val checked = requireNotNull(parameters[ToggleableStateKey]) {
+//            "This action should only be called in response to toggleable events"
+//        }
+        updateAppWidgetState(context, PreferencesGlanceStateDefinition, glanceId) {
+            it.toMutablePreferences()
+                .apply { set(booleanPreferencesKey(toggledString), checked) }
+        }
+        TodoListGlanceWidget().update(context, glanceId)
+    }
+}
+
+@Composable
+private fun CountToggled() {
+    val prefs = currentState<Preferences>()
+    val context = LocalContext.current
+    val checkedCount = groceryStringIds.filter {
+        prefs[booleanPreferencesKey(context.getString(it))] ?: false
```

Comment:
Can't we use `stringResource(id = it)` here?

Reply:
We can't use stringResource for Glance because we can't mix the Composables from the non glance version of the library.

Reply:
Oh, I see. Thanks!